### PR TITLE
8 date range refactor

### PR DESCRIPTION
**Pull request by peteburch223**
Date range edge cases addressed...although perhaps not in the most elegant way. Refactoring required.

### CODE REVIEW

File: app/controllers/spaces.rb [hunk 1]
```diff
@@ -2,9 +2,21 @@ class MakersBnB < Sinatra::Base
 
 
   get '/spaces' do
-    @spaces = Space.all(:availabledates => {:avail_date.gte => params[:from_date], :avail_date.lte => params[:to_date]},
-                        :fields => [:name,:description,:price],
-                        :order => nil)
+
+    date_from = Date.parse(params[:from_date]) if !params[:from_date].nil?
```
Comment (MuriloDalRi) [inline app/controllers/spaces.rb]: could be refactored to ``if params[:from_date]``

File: app/controllers/spaces.rb [hunk 2]
```diff
@@ -2,9 +2,21 @@ class MakersBnB < Sinatra::Base
...
+    date_to = Date.parse(params[:to_date]) if !params[:to_date].nil?
+    @spaces = []
+    avail = Hash.new(0)
+    if !!date_from && !!date_to
```
Comment (MuriloDalRi) [inline app/controllers/spaces.rb]: no need for !!

**Comment by MuriloDalRi** [comment]
:shipit:

**Comment by peteburch223** [comment]
:shipit:

**Pull request merged**